src/Psalm/Internal/Analyzer/Statements/Expression/ArrayAnalyzer.php

```diff
@@ -237,6 +241,38 @@ public static function analyze(
         return true;
     }
 
+    /**
+     * @param string|int $literal_array_key
+     * @return false|int
+     * @psalm-assert-if-false !numeric $literal_array_key
+     */
+    public static function getLiteralArrayKeyInt(
```

Comment:
I first thought we could replace it with something like:

```php
function convertIntishArrayKey(int|string $k): false|int {
    return is_int(key([$k => null])) ? $k : false;
}
```
but then realized that then target PHP version may have different rules.